### ISSUE TITLE
Code question in eval_sal.py

### ISSUE THREAD

**Issue by HIT-lcccc** [YuqiYang213/MLoRE#9]
The following code in evaluation/eval_sal.py line 35 to 37.
```
        assert preds.shape == target.shape

        if len(preds.shape) == len(target.shape) + 1:
            assert preds.shape[1] == 2
            # two class probabilites
            preds = nn.functional.softmax(preds, dim=1)[:, 1, :, :]
        else:
            # squash logits into probabilities
            preds = torch.sigmoid(preds)
```
The **assert** ensures the shape of _preds_ is same as _target_. In the following **if-else** branch, it will never go in 'if' branch. If so, the _preds_ will post-process with sigmoid. However, the 'preds' has been processed with softmax in get_output(utils/utils.py L39) before the SaliencyMeter. I have question whether the post-process of 'preds' is valid. 
The code may follow the Ye's [Multi-Task-Transformer](https://github.com/prismformore/Multi-Task-Transformer). But he didnt answer my question. I think the assert should be deleted. Would you help me solve the problem? Or I misunderstand the code.
Best regards.

**Comment by YuqiYang213** [comment]
Thank you for your question.
We also noticed this line after releasing our code. From our point of view, the post-process here with sigmoid is not valid and will result in a lower value than the true performance of saliency detection. 
We didn't delete this line in our repository to align with the performance reported in the paper. However, it is recommended to delete it and get the correct performance value.

**Comment by HIT-lcccc** [comment]
Thanks for your careful reply. Have a good day!
